File: chemist/ClientApp/src/app/services/ProductService.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { Product } from "../models/Product";

@Injectable()
export class ProductService {
    /**
     *
     */
    constructor(private http: HttpClient) {
        
    }

    public Insert(product:Product)
    {
        return this.http.post("/chemist/", product);
    }

    public Get(id:number)
    {
        return this.http.get<Product>('/chemist/'+id);
    }

    public GetRange(start:number, numberOfItems:number)
    {
        return this.http.get<Array<Product>>('/chemist/'+start+'/'+numberOfItems);
    }

    public Update(product:Product)
    {
        return this.http.put("/chemist/",product);
    }

    public Delete(product:Product)
    {
        return this.http.delete("/chemist/"+product.id);
    }

}

File: chemist/Services/ProductService.cs
using chemist.Data.Models;
using chemist.Repositories;

namespace chemist.Services
{
    public class ProductService:IProductService
    {
        private IProductRepository _productRepository;
        public ProductService(IProductRepository productRepository)
        {
            _productRepository = productRepository;
        }

        public void Delete(Product product)
        {
            _productRepository.Delete(product);
        }

        public Product Get(int productId)
        {
            return _productRepository.Get(productId);
        }

        public IEnumerable<Product> GetRange(int start, int numberOfItems)
        {
            return _productRepository.GetRange(start,numberOfItems);
        }

        public void Insert(Product product)
        {
            _productRepository.Insert(product);
        }

        public void Update(Product product)
        {
            _productRepository.Update(product);
        }
    }
}